contributors/devel/sig-instrumentation/trace-instrumentation.md

```diff
@@ -0,0 +1,71 @@
+## Instrumenting Kubernetes with Traces
+
+The following references and outlines general guidelines for trace instrumentation
+in Kubernetes components. Components are instrumented using the
+[OpenTelemetry Go client library](https://github.com/open-telemetry/opentelemetry-go).
+For non-Go components. [Libraries in other languages](https://opentelemetry.io/docs/languages/)
+are available.
+
+Traces are exposed via gRPC using the [OpenTelemetry Protocol](https://opentelemetry.io/docs/specs/otel/protocol/)
+(OTLP), which is open and well-understood by a wide range of third party
+applications and vendors in the cloud-native eco-system.
+
+The [general instrumentation advice](https://opentelemetry.io/docs/concepts/instrumentation/libraries/)
+from the OpenTelemetry documentation applies. This document reiterates common pitfalls and some
+Kubernetes specific considerations.
+
+### When to instrument
+
+While spans are sampled to avoid high costs, recording too many spans will
+force consumers to lower the sampling rate, and will "drown out" important
+spans. If your component has more than two or three nested spans, you are
+likely over-using trace instrumentation. Most trace instrumentation in
+Kubernetes components falls into one of two categories:
+
+1. Spans for incoming or outgoing network calls
+2. Spans when initiating new work, such as reconciling an object, which may result in network calls.
+
+For network-based telemetry, Kubernetes components should use OpenTelemetry
+instrumentation libraries for
+[HTTP](https://pkg.go.dev/go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp) and
+[gRPC](https://pkg.go.dev/go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc).
+
+**Note:** When creating spans at the start of reconciling an Object, only
+create the span changes are actually required. Avoid creating "empty" spans
+which simply compare the desired and actual state of an object without
+performing any real work, or making any network requests.
+
+### Configuration and Setup
+
+Kubernetes components should expose a flag, `--tracing-config-file`, which accepts a
+[TracingConfiguration](https://github.com/kubernetes/kubernetes/blob/cb989b84c82ab8c695d02c9a1866bc27de37caec/staging/src/k8s.io/component-base/tracing/api/v1/types.go)
```

Comment:
I would suggest linking the latest k8s docs rather than a single commit from the v1 types, i.e. https://kubernetes.io/docs/reference/config-api/apiserver-config.v1beta1/#apiserver-k8s-io-v1beta1-TracingConfiguration

Reply:
done